### PR TITLE
rgw/notifications/lc: handle publish_commit() failures as warnings

### PR DESCRIPTION
**Pull request by yuvalif**
Fixes: https://tracker.ceph.com/issues/63859

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by yuvalif** [comment]
jenkins test windows

File: src/rgw/rgw_lc.cc [hunk 1]
```diff
@@ -1366,13 +1366,13 @@ class LCOpAction_Transition : public LCOpAction {
...
+      int publish_ret = notify->publish_commit(oc.dpp, obj->get_obj_size(),
 				    ceph::real_clock::now(),
 				    obj->get_attrs()[RGW_ATTR_ETAG].to_str(),
 				    version_id);
-      if (ret < 0) {
-	ldpp_dout(oc.dpp, 1) <<
-	  "ERROR: notify publish_commit failed, with error: " << ret << dendl;
+      if (publish_ret < 0) {
+	ldpp_dout(oc.dpp, 5) <<
+	  "WARNING: notify publish_commit failed, with error: " << ret << dendl;
```
Comment (thotz) [inline src/rgw/rgw_lc.cc:1375]:
```suggestion
	  "WARNING: notify publish_commit failed, with error: " << publish_ret << dendl;
```

File: src/rgw/rgw_lc.cc [hunk 2]
```diff
@@ -860,13 +860,13 @@ int RGWLC::handle_multipart_expiration(rgw::sal::Bucket* target,
...
+        int publish_ret = notify->publish_commit(
             this, sal_obj->get_obj_size(), ceph::real_clock::now(),
             sal_obj->get_attrs()[RGW_ATTR_ETAG].to_str(),
 						version_id);
-        if (ret < 0) {
-          ldpp_dout(wk->get_lc(), 1)
-              << "ERROR: notify publish_commit failed, with error: " << ret
+        if (publish_ret < 0) {
+          ldpp_dout(wk->get_lc(), 5)
+              << "WARNING: notify publish_commit failed, with error: " << ret
```
Comment (thotz) [inline src/rgw/rgw_lc.cc:869]:
```suggestion
              << "WARNING: notify publish_commit failed, with error: " << publish_ret
```

File: src/rgw/rgw_lc.cc [hunk 3]
```diff
@@ -574,12 +574,12 @@ static int remove_expired_obj(
...
-    ret = notify->publish_commit(dpp, obj_state->size,
+    int publish_ret = notify->publish_commit(dpp, obj_state->size,
 				 ceph::real_clock::now(),
 				 obj_state->attrset[RGW_ATTR_ETAG].to_str(),
 				 version_id);
-    if (ret < 0) {
-      ldpp_dout(dpp, 1) << "ERROR: notify publish_commit failed, with error: "
+    if (publish_ret < 0) {
+      ldpp_dout(dpp, 5) << "WARNING: notify publish_commit failed, with error: "
 												<< ret << dendl;
```
Comment (thotz) [inline src/rgw/rgw_lc.cc:583]:
```suggestion
												<< publish_ret << dendl;
```

**Comment by yuvalif** [comment]
jenkins test make check

**Review by dang — APPROVED** [review]

**Comment by yuvalif** [comment]
teuthology passing: https://pulpito.ceph.com/yuvalif-2024-01-04_16:12:17-rgw:notifications-wip-yuval-test-distro-default-smithi/

**Pull request merged**